pandas/tests/plotting/test_datetimelike.py

```diff
@@ -44,6 +44,10 @@ def teardown_method(self, method):
         tm.close()
 
     @pytest.mark.slow
+    @pytest.mark.filterwarnings(
+        "ignore:Converting to PeriodArray/Index representation "
+        "will drop timezone information."
+    )
```

Comment:
any particular reason why youre using filterwarnings instead of assert_produces_warning here?

Reply:
Yes. It turns out that the warning only appears for UTC-like timezones. This will also work:
```python
        if (
            tz in ["UTC", pytz.UTC, timezone.utc]
            or isinstance(tz, dateutil.tz.tz.tzutc)
        ):
            # Converting to PeriodArray/Index representation will drop timezone
            # information.
            with tm.assert_produces_warning(UserWarning):
                _check_plot_works(ts.plot)
        else:
            _check_plot_works(ts.plot)
```